src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,309 @@
+                    else
+                    {
+                        return Runtime::GetPathTo(Runtime::PathName::PortableAppMachineRootX64);
+                    }
+                }
+            }
+            else
+            {
+                std::filesystem::path portableAppUserRoot = Settings::User().Get<Settings::Setting::PortableAppUserRoot>();
+                if (!portableAppUserRoot.empty())
+                {
+                    return portableAppUserRoot;
+                }
+                else
+                {
+                    return Runtime::GetPathTo(Runtime::PathName::PortableAppUserRoot);
+                }
+            }
+        }
+
+        std::filesystem::path GetPortableLinksLocation(Manifest::ScopeEnum& scope, Utility::Architecture& arch)
+        {
+            if (scope == Manifest::ScopeEnum::Machine)
+            {
+                if (arch == Utility::Architecture::X86)
+                {
+                    return Runtime::GetPathTo(Runtime::PathName::PortableAppMachineRootX86);
+                }
+                else
+                {
+                    return Runtime::GetPathTo(Runtime::PathName::PortableLinksMachineLocationX64);
+                }
+            }
+            else
+            {
+                return Runtime::GetPathTo(Runtime::PathName::PortableLinksUserLocation);
+            }
+        }
+
+        Manifest::AppsAndFeaturesEntry GetAppsAndFeaturesEntryForPortableInstall(std::vector<AppInstaller::Manifest::AppsAndFeaturesEntry> appsAndFeaturesEntries, AppInstaller::Manifest::Manifest& manifest)
+        {
+            AppInstaller::Manifest::AppsAndFeaturesEntry appsAndFeaturesEntry;
+            std::string packageId = manifest.Id;
+            std::string displayVersion = manifest.Version;
+            std::string displayName = manifest.DefaultLocalization.Get<Manifest::Localization::PackageName>();
+            std::string publisher = manifest.DefaultLocalization.Get<Manifest::Localization::Publisher>();
+
+            if (appsAndFeaturesEntries.empty())
+            {
+                appsAndFeaturesEntry = {};
+            }
+            else
+            {
+                appsAndFeaturesEntry = appsAndFeaturesEntries[0];
+            }
+
+            if (appsAndFeaturesEntry.DisplayName.empty())
+            {
+                appsAndFeaturesEntry.DisplayName = displayName;
+            }
+            if (appsAndFeaturesEntry.DisplayVersion.empty())
+            {
+                appsAndFeaturesEntry.DisplayVersion = displayVersion;
+            }
+            if (appsAndFeaturesEntry.Publisher.empty())
+            {
+                appsAndFeaturesEntry.Publisher = publisher;
+            }
+            if (appsAndFeaturesEntry.ProductCode.empty())
+            {
+                appsAndFeaturesEntry.ProductCode = packageId;
+            }
+
+            return appsAndFeaturesEntry;
+        }
+
+        void AppendExeExtension(std::string& value)
```

Comment:
Use `std::filesystem::path` if it is a path.

Reply:
Changed to std::filesystem::path